### PR TITLE
Fix parsing sql file

### LINKED ISSUE #14: Migration URL parsing is broken for versioned uberjars
Hello, Ivan! I hope you're doing well. I just ran into a bug with pg2 and I'm open to making a PR but I wanted to check with you on the preferred behavior. The bug is with the regex used for parsing migration files:

```clojure
(def RE_FILE #"(?i).*?(\d+)\.(.*?)\.?(prev|next|up|down)\.sql$")
```

When this is used within a versioned uberjar, it parses incorrectly. For example, this is the URL for a migration file in my uberjar:

```
"jar:file:/home/jeaye/projects/bayaz/target/uberjar/bayaz-0.1.0-SNAPSHOT-standalone.jar!/migrations/015.create-nicks-from-joins.next.sql"
```

The `0.1.0-SNAPSHOT` is what trips up the regex here. The result is that every single migration file will be parsed with id `0`, which leads to duplicate migration file errors. (example in [regex101](https://regex101.com/r/bMePKe/1)) For a Clojure example:

```clojure
> (re-matches RE_FILE "jar:file:/home/jeaye/projects/bayaz/target/uberjar/bayaz-0.1.0-SNAPSHOT-standalone.jar!/migrations/015.create-nicks-from-joins.next.sql")

["jar:file:/home/jeaye/projects/bayaz/target/uberjar/bayaz-0.1.0-SNAPSHOT-standalone.jar!/migrations/015.create-nicks-from-joins.next.sql"
 "0"
 "1.0-SNAPSHOT-standalone.jar!/migrations/015.create-nicks-from-joins"
 "next"]

```

If we changed the regex to require a `/` before the number, we can fix this issue. That should work correctly for both uberjar and non-uberjar cases (since the migration dir will always be before the file name). However, it would change the behavior of the `parse-name` fn, if your users are calling it with a file name having no `/` before it. The resulting regex would just be:

```clojure
(def RE_FILE #"(?i).*?/(\d+)\.(.*?)\.?(prev|next|up|down)\.sql$")
```

Ultimately, I'm not sure how you'd prefer to handle this. For now, I've changed my jar's naming scheme, but I imagine this is going to bite a lot of people who're running versioned uberjars.

### PR DESCRIPTION
**Pull request by igrishaev**
Fixes #14

### CODE REVIEW

File: pg-migration/src/pg/migration/core.clj [hunk 1]
```diff
@@ -166,7 +167,9 @@
   (when-let [[_ id-raw slug-raw direction-raw]
              (re-matches RE_FILE (-> url
                                      .getFile
```
Comment (igrishaev) [inline pg-migration/src/pg/migration/core.clj:169]: The key change: we parse not the entire file path but only the trailing file name, e.g.

`file://path/to/jar-1.2.3/resoures/123.foo.sql` -> `123.foo.sql`

File: pg-migration/test/pg/migration_test.clj [hunk 2]
```diff
@@ -123,7 +123,31 @@
...
+          (-> "003.hello.prev.sql"
+              io/file
+              io/as-url
+              mig/parse-url)]
+
+      (is (= {:id 3
+              :slug "hello"
+              :direction :prev}
+             (dissoc res :url)))))
+
```
Comment (igrishaev) [inline pg-migration/test/pg/migration_test.clj:139]: Does the result of this case look good? @jeaye

Comment (jeaye) [reply pg-migration/test/pg/migration_test.clj:139]: Looks great! I've tested this locally and it fixed the issue. Thank you!

After merging, if you wouldn't mind publishing a new release for this, that would be superb.

Comment (igrishaev) [reply pg-migration/test/pg/migration_test.clj:139]: Yep, will do tomorrow.

**Pull request merged**